### PR TITLE
Fix memory leaks in init and pool allocator

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

Fix remaining leaks found with AddressSanitizer. One in initialization, and one in the pool allocator.

## Expected Impact

Fix memory leaks.

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Have read and agree to the Yaksa CLA terms (https://github.com/pmodels/yaksa/wiki/Yaksa-Contributor-License-Agreement)

### CODE REVIEW

**Comment by raffenet** [comment]
I've got one more commit the fixes the remaining leaks. Will push and update description shortly.

File: src/util/yaksu_pool.c [hunk 1]
```diff
@@ -77,6 +77,7 @@ int yaksu_pool_free(yaksu_pool_s pool)
                 count++;
 
         pool_head->free_fn(tmp->slab);
+        pool_head->free_fn(tmp->elems);
```
Comment (pavanbalaji) [inline src/util/yaksu_pool.c:80]:
```suggestion
        free(tmp->elems);
```

Comment (pavanbalaji) [reply src/util/yaksu_pool.c:80]: I need to go back and see why my valgrind testing didn't catch any of these errors.  These are fairly obvious errors that valgrind should be able to catch.

Comment (raffenet) [reply src/util/yaksu_pool.c]: AddressSanitizer is a wonder. I think `libfabric` enables it be default for debug configurations. Probably a good idea for new projects.

Comment (pavanbalaji) [reply src/util/yaksu_pool.c]: I'll add it to yaksa's testing too.

**Pull request merged**